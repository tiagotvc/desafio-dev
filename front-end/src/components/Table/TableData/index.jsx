
import React from 'react';
import { mock } from './data-mock';

const TableTr = (props) => {
    const objData = Object.values(props.data);
        return (
            <tr>
                {objData.map((values) => {
                    return (
                        <td>{values}</td>
                    )
                })}
            </tr>
        )
}

export const TableData = () => {
    return (
        <tbody>
            {mock.map((data) =>  {
                return (
                    <TableTr data={data}/>
                )
            }           
        )}
        </tbody>
    )
}