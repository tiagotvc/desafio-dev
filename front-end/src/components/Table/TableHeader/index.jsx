import React from 'react';

const ths = ['Nome','DÃ©bito','Boleto',
'Financiamento','CrÃ©dito','Recebimento EmprÃ©stimo',
'Vendas','Recebimento TED','Recebimento DOC','Aluguel'];

export const TableHeader = () => {
    return (
        <tr id='1'>
            {ths.map((name, index) => {
                return (<th key={index}>{name}</th>)
            })}
        </tr>
    )
}