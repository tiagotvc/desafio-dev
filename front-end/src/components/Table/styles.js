import styled from 'styled-components';

export const Container = styled.div`
    display: flex;
    flex-direction: column;
    width: 50%;

    .table, td, th {
        border: 1px solid #ddd;
        text-align: left;
    }
`
