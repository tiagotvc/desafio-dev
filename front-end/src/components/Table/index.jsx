import React from 'react';
import * as Styled from './styles';

export const Table = (props) => {

    return (
        <></>
    )
}