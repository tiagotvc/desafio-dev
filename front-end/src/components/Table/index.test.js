import React from 'react';
import Table from './index';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

describe('Table ', () => {
    const data = {name:'tiago',debeti:'00'}
    it('renders correctly', () => {
      const tree = renderer
      .create(<Table data/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
      });
  });