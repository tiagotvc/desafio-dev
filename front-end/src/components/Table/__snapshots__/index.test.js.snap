// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table  renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 .table,
.c0 td,
.c0 th {
  border: 1px solid #ddd;
  text-align: left;
}

<div
  className="c0"
>
  <table>
    <thead>
      <tr
        id="1"
      >
        <th>
          Nome
        </th>
        <th>
          Débito
        </th>
        <th>
          Boleto
        </th>
        <th>
          Financiamento
        </th>
        <th>
          Crédito
        </th>
        <th>
          Recebimento Empréstimo
        </th>
        <th>
          Vendas
        </th>
        <th>
          Recebimento TED
        </th>
        <th>
          Recebimento DOC
        </th>
        <th>
          Aluguel
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
